File: Domain/Company.cs
namespace Domain;

public class Company
{
    public Guid Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public string Url { get; set; }
    public ICollection<AppUser> Representers { get; set; }
    public ICollection<Event> Events { get; set; }
}

File: client-app/src/app/models/Company.ts
import { Profile } from "./Profile";

export interface Company {
    id: string;
    title: string;
    url: string;
    description: string;
    representers: Profile[];
}

export class Company implements Company {
    constructor(init?: CompanyFormValues) {
        Object.assign(this, init);
    }
}

export class CompanyFormValues {
    id: string | undefined = undefined;
    title: string = '';
    url: string = '';
    description: string = '';
    representers: Profile[] = [];
}